### DISCUSSION TITLE
Support for application/x-www-form-urlencoded in request body UI playground

### DISCUSSION THREAD

**Discussion by mouhannad-sh** [scalar/scalar#654] [Q&A]
Hi There 👋🏻 

I'm thinking of supporting the `application/x-www-form-urlencoded ` body input, swagger UI already have this but I feel Scalar's UI elements are far better suited in terms of usability for this.

As you can see swagger ui treats those fields like the following 

<img width="780" alt="image" src="https://github.com/scalar/scalar/assets/18495740/f6134f21-cfad-49c6-b935-6bffe94df9cb">

Based on the existing codebase I can see that it's kind of supported-ish in here 

https://github.com/scalar/scalar/blob/00c01b56b37a278083ce1bc3b38714061d154703/packages/api-reference/src/helpers/getRequestFromOperation.ts#L147-L162

usually I'd use `URLSearchParams` to serialize body but I'm assuming that for the UI components to work properly we're gonna need a data structure that's familiar with it 

From the comments in the code I can see this:

```
// TODO: We have an object, but how do we get that kind of array from the object?
        // Don’t forget to include nested properties … :|
```
 
Looking at swagger UI and Postman, I think the usage of the grid when the contentType is `Content-Type: application/x-www-form-urlencoded ` might be a good idea.

Since I basically just joined the contributors list (thank you 🙌🏻 ) I'm wondering what would be the considerations in adding this capability, I think some technical perspective on this from the maintainers and/or community users would be nice before I jump into implementation 

Thanks in advance !

**Comment by marclave** [comment]
hi @mouhannad-sh ! we are working on adding multiple types soon ✨ will make an issue for this

**Reply by hanspagel** [reply]
Just for reference, the issue is here: https://github.com/scalar/scalar/issues/1852
